middle_end/flambda/simplify/expr_builder.ml

```diff
@@ -119,6 +119,7 @@ let create_singleton_let uacc (bound_var : VB.t) defining_expr
     in
     let uacc =
       UA.with_name_occurrences uacc ~name_occurrences:free_names_of_let
+      |> UA.increment_size (Code_size.increase_due_to_let_expr ~size_of_defining_expr)
```

Comment:
Here, if the let-binding is a phantom one, we should not increase the size at all.